src/erldantic_json_schema.erl

```diff
@@ -0,0 +1,388 @@
+    to_schema_no_pt(Module, {record, RecordName}).
+
+%% INTERNAL
+
+-spec to_schema_no_pt(Module :: module(), TypeRef :: erldantic:ed_type_reference()) ->
+                         {ok, Schema :: map()} | {error, [erldantic:error()]}.
+to_schema_no_pt(Module, TypeRef) ->
+    try
+        TypeInfo = erldantic_module_types:get(Module),
+        to_schema(TypeInfo, TypeRef)
+    catch
+        error:Reason ->
+            {error,
+             [#ed_error{type = no_match,
+                        location = [Module],
+                        ctx = #{reason => Reason, error_type => module_extraction_failed}}]}
+    end.
+
+-spec to_schema(erldantic:type_info(), erldantic:ed_type_or_ref()) ->
+                   {ok, Schema :: map()} | {error, [erldantic:error()]}.
+to_schema(TypeInfo, Type) ->
+    do_to_schema(TypeInfo, Type).
+
+-spec do_to_schema(TypeInfo :: erldantic:type_info(),
+                   Type :: erldantic:ed_type_or_ref()) ->
+                      {ok, Schema :: map()} | {error, [erldantic:error()]}.
+%% Simple types
+do_to_schema(_TypeInfo, #ed_simple_type{type = integer}) ->
+    {ok, #{type => <<"integer">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = string}) ->
+    {ok, #{type => <<"string">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = boolean}) ->
+    {ok, #{type => <<"boolean">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = number}) ->
+    {ok, #{type => <<"number">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = float}) ->
+    {ok, #{type => <<"number">>, format => <<"float">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = atom}) ->
+    {ok, #{type => <<"string">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = binary}) ->
+    {ok, #{type => <<"string">>, format => <<"binary">>}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = nonempty_binary}) ->
+    {ok,
+     #{type => <<"string">>,
+       format => <<"binary">>,
+       minLength => 1}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = nonempty_string}) ->
+    {ok, #{type => <<"string">>, minLength => 1}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = pos_integer}) ->
+    {ok, #{type => <<"integer">>, minimum => 1}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = non_neg_integer}) ->
+    {ok, #{type => <<"integer">>, minimum => 0}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = neg_integer}) ->
+    {ok, #{type => <<"integer">>, maximum => -1}};
+do_to_schema(_TypeInfo, #ed_simple_type{type = term}) ->
+    {ok, #{}};  % any type
+%% Range types
+do_to_schema(_TypeInfo,
+             #ed_range{type = integer,
+                       lower_bound = Min,
+                       upper_bound = Max}) ->
+    {ok,
+     #{type => <<"integer">>,
+       minimum => Min,
+       maximum => Max}};
+%% Literal types
+do_to_schema(_TypeInfo, #ed_literal{value = undefined}) ->
+    {ok, #{enum => [null]}};
+do_to_schema(_TypeInfo, #ed_literal{value = Value}) ->
+    {ok, #{enum => [Value]}};
```

Comment:
Hmm, most literal values doesn't translate well to json. How have I done it in erldantic_json?